unittests/test_containers.py

```diff
@@ -157,8 +186,67 @@ def test_prepare_command(container_platform, expected_cmd_prepare):
 
 
 def test_run_opts(container_platform, expected_cmd_run_opts):
-    container_platform.commands = ['cmd']
     container_platform.mount_points = [('/path/one', '/one')]
-    container_platform.workdir = '/stagedir'
     container_platform.options = ['--foo', '--bar']
     assert container_platform.launch_command() == expected_cmd_run_opts
+
+
+@pytest.fixture(params=[
+    'Docker', 'Singularity', 'Sarus', 'Shifter'])
+def container_variant_deprecated(request):
+    return request.param
+
+
+@pytest.fixture
+def platform_deprecated(container_variant_deprecated):
```

Comment:
I don't understand what do you want to achieve with this here?

Reply:
I wanted to separate the tests related to deprecated fields so that it is easier to remove afterwards.

Reply:
Couldn't you simply add it to the normal test and wrap the deprecated operations with the following?

```python
with pytest.warns(ReframeDeprecationWarning):
    deprecated_op()
```